registered_span.go

```diff
@@ -0,0 +1,1056 @@
+			"http.path_tpl": yes,
+			"http.host":     yes,
+			"http.protocol": yes,
+			"http.error":    yes,
+		}
+	case RPCServerSpanType, RPCClientSpanType:
+		return map[string]struct{}{
+			"rpc.host":      yes,
+			"rpc.port":      yes,
+			"rpc.call":      yes,
+			"rpc.call_type": yes,
+			"rpc.flavor":    yes,
+			"rpc.error":     yes,
+		}
+	case KafkaSpanType:
+		return map[string]struct{}{
+			"kafka.service": yes,
+			"kafka.access":  yes,
+		}
+	case GCPStorageSpanType:
+		return map[string]struct{}{
+			"gcs.op":                 yes,
+			"gcs.bucket":             yes,
+			"gcs.object":             yes,
+			"gcs.entity":             yes,
+			"gcs.range":              yes,
+			"gcs.sourceBucket":       yes,
+			"gcs.sourceObject":       yes,
+			"gcs.destinationBucket":  yes,
+			"gcs.destinationObject":  yes,
+			"gcs.numberOfOperations": yes,
+			"gcs.projectId":          yes,
+			"gcs.accessId":           yes,
+		}
+	case GCPPubSubSpanType:
+		return map[string]struct{}{
+			"gcps.projid": yes,
+			"gcps.op":     yes,
+			"gcps.top":    yes,
+			"gcps.sub":    yes,
+			"gcps.msgid":  yes,
+		}
+	case AWSLambdaEntrySpanType:
+		return map[string]struct{}{
+			"lambda.arn":                    yes,
+			"lambda.name":                   yes,
+			"lambda.version":                yes,
+			"lambda.trigger":                yes,
+			"cloudwatch.events.id":          yes,
+			"cloudwatch.events.resources":   yes,
+			"cloudwatch.logs.group":         yes,
+			"cloudwatch.logs.stream":        yes,
+			"cloudwatch.logs.decodingError": yes,
+			"cloudwatch.logs.events":        yes,
+			"s3.events":                     yes,
+			"sqs.messages":                  yes,
+		}
+	case AWSS3SpanType:
+		return map[string]struct{}{
+			"s3.region": yes,
+			"s3.op":     yes,
+			"s3.bucket": yes,
+			"s3.key":    yes,
+			"s3.error":  yes,
+		}
+	case AWSSQSSpanType:
+		return map[string]struct{}{
+			"sqs.sort":  yes,
+			"sqs.queue": yes,
+			"sqs.type":  yes,
+			"sqs.group": yes,
+			"sqs.size":  yes,
+			"sqs.error": yes,
+		}
+	case AWSSNSSpanType:
+		return map[string]struct{}{
+			"sns.topic":   yes,
+			"sns.target":  yes,
+			"sns.phone":   yes,
+			"sns.subject": yes,
+			"sns.error":   yes,
+		}
+	case AWSDynamoDBSpanType:
+		return map[string]struct{}{
+			"dynamodb.table": yes,
+			"dynamodb.op":    yes,
+			"dynamodb.error": yes,
+		}
+	default:
+		return nil
+	}
+}
+
+// HTTPSpanData represents the `data` section of an HTTP span sent within an OT span document
+type HTTPSpanData struct {
+	SpanData
+	Tags HTTPSpanTags `json:"http"`
+
+	clientSpan bool
+}
+
+// NewHTTPSpanData initializes a new HTTP span data from tracer span
+func NewHTTPSpanData(span *spanS) HTTPSpanData {
+	data := HTTPSpanData{
+		SpanData: NewSpanData(span, RegisteredSpanType(span.Operation)),
+		Tags:     NewHTTPSpanTags(span),
+	}
+
+	kindTag := span.Tags[string(ext.SpanKind)]
+	data.clientSpan = kindTag == ext.SpanKindRPCClientEnum || kindTag == (ext.SpanKindRPCClientEnum)
```

Comment:
It seems there is a typo

```suggestion
	data.clientSpan = kindTag == ext.SpanKindRPCClientEnum || kindTag == string(ext.SpanKindRPCClientEnum)
```

Reply:
Well spotted, I believe there should be no type cast though

Reply:
My bad, I misread this change :) You're right!